source/common/res/features/budget-to-goal/main.js

```diff
@@ -0,0 +1,163 @@
+(function poll() {
+  if (typeof ynabToolKit !== 'undefined' && ynabToolKit.pageReady === true) {
+    ynabToolKit.budgetBalanceToGoal = (function () {
+      return {
+        budgetView: ynab.YNABSharedLib.getBudgetViewModel_AllBudgetMonthsViewModel()._result,
+
+        invoke() {
+          let categories = ynabToolKit.budgetBalanceToGoal.getCategories();
+          let categoryName = ynabToolKit.budgetBalanceToGoal.getInspectorName();
+          let masterCategoryViewId = $('ul.is-checked').prevAll('ul.is-master-category').attr('id');
+          let masterCategory = ynabToolKit.shared.getEmberView(masterCategoryViewId).get('data');
+          let masterCategoryId = masterCategory.get('categoryId');
+
+          categories.forEach(function (f) {
+            if (f.name === categoryName && f.masterCategoryId === masterCategoryId) {
+              ynabToolKit.budgetBalanceToGoal.updateInspectorButton(f);
+
+              return false;
+            }
+          });
+        },
+
+        observe(changedNodes) {
+          console.log(changedNodes);
+          if (changedNodes.has('budget-inspector-goals')) {
+            // In theory, if there's a 'budget-inspector-goals' classed node, then the user is or has changed the goal.
+            // However, there seems to be some disconnect between this event and when YNAB updates the values.
```

Comment:
Two options:

- You could probably watch for the 'Goal Target' button text to change to trigger your invoke here.
- If that doesn't work, then you can likely listen to the actual data storage object for changes, but that'd be harder than the above, and I'd need to have a play to see if I can figure out get that to work, so try the above first and let me know if that doesn't work.